### ISSUE TITLE
PIN accessible in plain text?

### ISSUE THREAD

**Issue by jbrepogmailcom** [mycelium-com/wallet-android#617]
Hello. In my phone, settings show that it is using File Based Encryption (Android 10). Therefore I assume that data partition is completely encrypted. However, when I create backup with phone's internal backup software (Hisense), the application data from data partition are TARred without encryption and placed to public storage. There I can extract mycelium data and I see that in settings.xml the PIN is stored as unencrypted.

I see this as security flaw - anyone who successfully gets access to mycelium data can extract the PIN.

**Comment by Giszmo** [comment]
More worrying than the pin would be the masterseed which I assume is in the backup, too. Please do never backup any bitcoin wallet in any other way than on paper.

That said, the app should [not allow backups](https://github.com/mycelium-com/wallet-android/blob/master/mbw/src/main/AndroidManifest.xml#L33) but as this is a flag your OS can ignore, it might be a Hisense "feature" to allow you to pull stuff out of protected storage.

Please describe exactly how you make the backup and if your phone is rooted.

**Comment by jbrepogmailcom** [comment]
Yes I believe that Hisense backup does not do android backup (to .ab file), so backup flag in manifest is not relevant. Instead, Hisense has internal system application "Backup" that allows to pick any application and then it creates tar with apk file and with application data from data partition. This TAR is then saved anywhere the user choses, in public internal memory or on SD card. TAR files is not encrypted.

I would like to believe that if I save TAR backup to "publicly" accessible internal memory (where Downloads folder is), it is still protected in locked phone because it uses Android 10 and File Based Encryption. But I am not sure how internal memory is encrypted because I cannot test it on locked phone.

So only own mycelium backup seems to be safe. However, that one does not save notes in transactions, right?

**Comment by Giszmo** [comment]
> So only own mycelium backup seems to be safe. However, that one does not save notes in transactions, right?

You cannot backup labels, that is right.

If you expect the backup to be encrypted, I wonder how you expect it to work practically. Who has the key? If you have the backup and move to a new phone, would you expect Hisense to have the key? What if you backup the blockchain.com wallet, the most used Bitcoin wallet with somewhat public source? I bet you will find key material, too.

**If you had money in your wallet prior to making this backup, please move your funds to a completely new wallet. Funds backed up with the protected files being exposed in "sd card" may disappear at any time.**
